pyiron/sphinx/base.py

```diff
@@ -634,26 +897,32 @@ def set_empty_states(self, n_empty_states=None):
         Function to set the number of empty states.
 
         Args:
-            n_empty_states (int/None): Number of empty states. If None, sets it to 'auto'.
+            n_empty_states (int/None): Number of empty states.
+            If None, sets it to 'auto'.
 
         Comments:
-            If this number is too low, the algorithm will not be able to able to swap wave functions
-            near the chemical potential. If the number is too high, computation time will be wasted
-            for the higher energy states and potentially lead to a memory problem.
+            If this number is too low, the algorithm will not be
+            able to able to swap wave functions near the chemical
+            potential. If the number is too high, computation time
+            will be wasted for the higher energy states and
+            potentially lead to a memory problem.
 
-            In contrast to VASP, this function sets only the number of empty states and not the number of
-            total states.
+            In contrast to VASP, this function sets only the number
+            of empty states and not the number of total states.
 
-            The default value is 0.5*NIONS+3 for non-magnetic systems and 1.5*NIONS+3 for magnetic systems
+            The default value is 0.5*NIONS+3 for non-magnetic systems
+            and 1.5*NIONS+3 for magnetic systems
         """
         if n_empty_states is None:
             self.input["EmptyStates"] = "auto"
         else:
             if n_empty_states < 0:
-                raise ValueError("Number of empty states must be greater than 0")
+                raise ValueError(
+                    "Number of empty states must be greater than 0"
+                    )
             self.input["EmptyStates"] = n_empty_states
 
-    def _set_kpoints(
+    def set_kpoints(
```

Comment:
@sudarsan-surendralal Can you comment on this? I though that the codes should implement the `_set_kpoints` function while `set_kpoints` is only defined in `GenericDFT`.

Reply:
I agree, but I can't remember what was the argument.

Reply:
I guess the main reason is that there are checks implemented on the generic level which should be available in every code https://github.com/pyiron/pyiron/blob/master/pyiron/dft/job/generic.py#L206 
Another and maybe more fancy way of implementing this would be using a decorator an example could be found here https://realpython.com/primer-on-python-decorators/#simple-decorators then we could have an `@set_kpoints` decorator which modifies the `set_kpoints` function in the specific code accordingly. Still this should not be part of this pull request.

Reply:
Did we decide this? Because for the VASP module, both `set_kpoints` and `_set_kpoints` are implemented at the base class. But I agree, we should discuss this in a separate pull request.

Reply:
Okay I reverted it to `_set_kpoints` as of 7b05294c, and created #612